physx-sys/src/lib.rs

```diff
@@ -171,6 +171,48 @@ pub const fn version(major: u32, minor: u32, patch: u32) -> u32 {
 pub type CollisionCallback =
     unsafe extern "C" fn(*mut c_void, *const PxContactPairHeader, *const PxContactPair, count: u32);
 
+pub type TriggerCallback = unsafe extern "C" fn(*mut c_void, *const PxTriggerPair, count: u32);
+
+pub type ConstraintBreakCallback =
+    unsafe extern "C" fn(*mut c_void, *const PxConstraintInfo, count: u32);
+
+pub type WakeSleepCallback =
+    unsafe extern "C" fn(*mut c_void, *const *const PxActor, count: u32, bool);
+
+pub type AdvanceCallback =
+    unsafe extern "C" fn(*mut c_void, *const *const PxRigidBody, *const PxTransform, count: u32);
+
+#[repr(C)]
+pub struct SimulationEventCallbackInfo {
+    pub collision_callback: Option<CollisionCallback>, // Yes, option<funcptr> becomes a nullable C pointer
+    pub collision_user_data: *mut c_void,
+    pub trigger_callback: Option<TriggerCallback>,
+    pub trigger_user_data: *mut c_void,
+    pub constraint_break_callback: Option<ConstraintBreakCallback>,
+    pub constraint_break_user_data: *mut c_void,
+    pub wake_sleep_callback: Option<WakeSleepCallback>,
+    pub wake_sleep_user_data: *mut c_void,
+    pub advance_callback: Option<AdvanceCallback>,
+    pub advance_user_data: *mut c_void,
+}
```

Comment:
I'd maybe add some description here of these fields and maybe just space them out a bit! :-)

Reply:
Also, what's the semantic difference between the context being a ptr and a nullable function ptr? I.e., could we represent the latter as `Option<NonNull<Any>>`? Or if the context always must exist, just `NonNull<Any>`? Or a box? :-) Trying to think if/how we can be more clear about semantics.

Reply:
Clarified it with a new comment: 

// Function pointers in Rust are normally not nullable (which is why they don't require unsafe to call)
// but we need them to be, so we simply wrap them in Option<>. An Option<funcptr> is luckily represented
// by the compiler as a simple pointer with null representing None, so this is compatible with the C struct.